src/contract-service.js

```diff
@@ -104,13 +110,28 @@ class ContractService {
     }
   }
 
-  async deployed(contract) {
+  async deployed(contract, addrs) {
     const net = await this.web3.eth.net.getId()
-    const addrs = contract.networks[net]
-    return new this.web3.eth.Contract(
-      contract.abi,
-      addrs ? addrs.address : null
-    )
+    let storedAddress = contract.networks[net] && contract.networks[net].address
+    addrs = addrs || storedAddress || null
+    return new this.web3.eth.Contract(contract.abi, addrs)
+  }
+
+  async deploy(contract, args, options) {
+    let deployed = await this.deployed(contract)
+    let transaction = await new Promise((resolve, reject) => {
+      deployed
+        .deploy({
+          data: contract.bytecode,
+          arguments: args
+        })
+        .send(options)
+        .on("receipt", receipt => {
```

Comment:
Should we listen for the "confirmation" event instead?

Reply:
@nick I'm actually not that familiar with this API. I just looked at what we're doing in `ResourceBase.contractFn` with `waitTransactionFinished` and did the same thing here.

https://github.com/OriginProtocol/origin-js/blob/master/src/ResourceBase.js#L24